pkg/dashboard/resource/frontendspec.go

```diff
@@ -76,6 +82,29 @@ func (fesr *frontendSpecResource) getFrontendSpec(request *http.Request) (*restf
 	}, nil
 }
 
+func (fesr *frontendSpecResource) getDefaultFunctionConfig() map[string]interface{} {
+	one := 1
+	defaultWorkerAvailabilityTimeoutMilliseconds := trigger.DefaultWorkerAvailabilityTimeoutMilliseconds
+	defaultFunctionSpec := functionconfig.Spec{
+		MinReplicas:             &one,
+		MaxReplicas:             &one,
+		ReadinessTimeoutSeconds: abstract.DefaultReadinessTimeoutSeconds,
+		TargetCPU:               functionres.DefaultTargetCPU,
+		Triggers: map[string]functionconfig.Trigger{
+
+			// notice that this is a mapping between trigger kind and its default values
+			"http": {
+				WorkerAvailabilityTimeoutMilliseconds: &defaultWorkerAvailabilityTimeoutMilliseconds,
+			},
+			"cron": {
+				WorkerAvailabilityTimeoutMilliseconds: &defaultWorkerAvailabilityTimeoutMilliseconds,
+			},
+		},
```

Comment:
No other defaults we need to inject here?

Reply:
no